File: Lista de Exercicio Paradigmas/Lista Java/Ex2.java
public class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private int velocidade;

    public Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0; 
    }

    public String detalhes() {
        return "Marca: " + this.marca + ", Modelo: " + this.modelo + ", Ano: " + this.ano;
    }

    public void acelerar(int incremento) {
        this.velocidade += incremento;
        System.out.println("O carro acelerou para " + this.velocidade + " km/h.");
    }

    public void frear(int decremento) {
        this.velocidade -= decremento;
        if (this.velocidade < 0) {
            this.velocidade = 0; 
        }
        System.out.println("O carro freou para " + this.velocidade + " km/h.");
    }

    public void exibirVelocidade() {
        System.out.println("A velocidade atual é " + this.velocidade + " km/h.");
    }

    public static void main(String[] args) {
        Carro carro1 = new Carro("Toyota", "Corolla", 2020);
        Carro carro2 = new Carro("Honda", "Civic", 2019);
        Carro carro3 = new Carro("Ford", "Fiesta", 2021);

        System.out.println(carro1.detalhes());
        System.out.println(carro2.detalhes());
        System.out.println(carro3.detalhes());

        carro1.acelerar(50);
        carro1.exibirVelocidade();
        carro1.frear(20);
        carro1.exibirVelocidade();
    }
}

File: Lista de Exercicio Paradigmas/Lista Python/Ex2.py
class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.velocidade = 0  

    def detalhes(self):
        return f'Marca: {self.marca}, Modelo: {self.modelo}, Ano: {self.ano}'

    def acelerar(self, incremento):
        self.velocidade += incremento
        print(f'O carro acelerou para {self.velocidade} km/h.')

    def frear(self, decremento):
        self.velocidade -= decremento
        if self.velocidade < 0:
            self.velocidade = 0  
        print(f'O carro freou para {self.velocidade} km/h.')

    def exibir_velocidade(self):
        print(f'A velocidade atual é {self.velocidade} km/h.')

carro1 = Carro('Toyota', 'Corolla', 2020)
carro2 = Carro('Honda', 'Civic', 2019)
carro3 = Carro('Ford', 'Fiesta', 2021)

print(carro1.detalhes())
print(carro2.detalhes())
print(carro3.detalhes())

carro1.acelerar(50)
carro1.exibir_velocidade()
carro1.frear(20)
carro1.exibir_velocidade()